processor/src/test/java/org/mapstruct/ap/test/nestedtargetproperties/NestedTargetPropertiesTest.java

```diff
@@ -51,11 +50,11 @@
 @RunWith(AnnotationProcessorTestRunner.class)
 public class NestedTargetPropertiesTest {
 
-    @Rule
-    public GeneratedSource generatedSource = new GeneratedSource().addComparisonToFixtureFor(
-        ChartEntryToArtist.class,
-        ChartEntryToArtistUpdate.class
-    );
+//    @Rule
+//    public GeneratedSource generatedSource = new GeneratedSource().addComparisonToFixtureFor(
+//        ChartEntryToArtist.class,
+//        ChartEntryToArtistUpdate.class
+//    );
```

Comment:
@navpil the problem are the `HashMap`(s) used in the `NestedTargetPropertyMappingHolder`. If you change them to `LinkedHashMap`(s) then all is good.